packages/api/tests/lib/test-granules.js

```diff
@@ -91,3 +282,221 @@ test('getGranuleProductVolume() returns correct product volume', (t) => {
     0
   );
 });
+
+test.serial('deleteGranuleAndFiles() removes a granule from PG and Dynamo', async (t) => {
+  const { newPgGranule, newDynamoGranule } = await createGranuleAndFiles(
+    t.context.knex,
+    t.context.testPgCollection.cumulus_id,
+    false
+  );
+
+  await deleteGranuleAndFiles({
+    knex: t.context.knex,
+    dynamoGranule: newDynamoGranule,
+    pgGranule: newPgGranule,
+    granuleModelClient: granuleModel,
+  });
+
+  // Check Dynamo and RDS. The granule should have been removed from both.
+  await t.throwsAsync(
+    granuleModel.get({ granuleId: newDynamoGranule.granuleId }),
+    { instanceOf: RecordDoesNotExist }
+  );
+
+  await t.throwsAsync(
+    granulePgModel.get(t.context.knex, { granule_id: newPgGranule.granule_id }),
+    { instanceOf: RecordDoesNotExist }
+  );
+});
+
+test.serial('deleteGranuleAndFiles() removes files from PG and S3', async (t) => {
+  const { newPgGranule, newDynamoGranule, files } = await createGranuleAndFiles(
+    t.context.knex,
+    t.context.testPgCollection.cumulus_id,
+    false
+  );
+
+  await deleteGranuleAndFiles({
+    knex: t.context.knex,
+    dynamoGranule: newDynamoGranule,
+    pgGranule: newPgGranule,
+    granuleModelClient: granuleModel,
+  });
+
+  // verify the files are deleted from S3. No need to check the Postgres files.
+  // If the granule was successfully deleted, the postgres
+  // files will have been as well. Files have a fk which would
+  // prevent the granule from being deleted if files referencing it
+  // still exist.
+  /* eslint-disable no-await-in-loop */
+  for (let i = 0; i < files.length; i += 1) {
+    const file = files[i];
+    t.false(await s3ObjectExists({ Bucket: file.bucket, Key: file.key }));
```

Comment:
Where is the check that the files no longer exist in PG?

Reply:
(aside) The comment explicitly implies that:

  // No need to check the Postgres files.
  // If the granule was successfully deleted, the postgres
(/aside) 

Relying on the database schema (can't delete granule if files are present) may be fair in an phase 1 API test, however the title should be changed now that you point this out.

Reply:
Ah, if the granule in PG was deleted, then the files must have been deleted given the FK constraints? I suppose that's fair. Some part of me wonders if it's still worth checking explicitly, but not gonna hold this PR up on that

Reply:
Yeah, that was the idea but typing this response is taking way more time that just checking the PG files did 😄 

26afa2ab08540d84652935532847a8a338f499e2